src/lambda/commands/deploySamApplication.ts

```diff
@@ -13,137 +13,200 @@ import * as nls from 'vscode-nls'
-                outputTemplatePath,
-                stackName,
-                region,
-                parameterOverrides,
-                invoker,
-                profile
-            )
-            // Deploying can take a very long time for Python Lambda's with native dependencies so user needs feedback
-            restParams.outputChannel.appendLine(localize(
-                'AWS.samcli.deploy.workflow.stackName.initiated',
-                'Deploying SAM Application to CloudFormation Stack: {0} with profile: {1}',
-                stackName, profile
-            ))
-            await deployInvocation.execute()
-
-            const msg = localize(
-                'AWS.samcli.deploy.workflow.success',
-                'Successfully deployed SAM Application to CloudFormation Stack: {0} with profile: {1}',
-                stackName, profile
-            )
-            restParams.outputChannel.appendLine(msg)
-            // TODO: Is this the right way to provide this feedback?
-            vscode.window.showInformationMessage(msg)
-        } catch (err) {
-            let msg = localize(
-                'AWS.samcli.deploy.workflow.error',
-                'Failed to deploy SAM application. Error while {0}: {1}',
-                stage, String(err)
-            )
-            // tslint:disable-next-line:max-line-length
-            // detect error message from https://github.com/aws/aws-cli/blob/4ff0cbacbac69a21d4dd701921fe0759cf7852ed/awscli/customizations/cloudformation/exceptions.py#L42
-            // and append region to assist in troubleshooting the error
-            // (command uses CLI configured value--users that don't know this and omit region won't see error)
-            if (msg.includes(`aws cloudformation describe-stack-events --stack-name ${args.stackName}`)) {
-                msg += ` --region ${args.region}`
-                if (profile) {
-                    msg += ` --profile ${profile}`
-                }
+        params.channelLogger.channel.appendLine(localize(
+            'AWS.samcli.deploy.workflow.init',
+            'Building SAM Application...'
+        ))
+        await build.execute()
+
+        stage = 'packaging'
+        const packageInvocation = new SamCliPackageInvocation(
+            buildTemplatePath,
+            packageTemplatePath,
+            params.deployParameters.packageBucketName,
+            params.invoker,
+            params.deployParameters.region,
+            params.deployParameters.profile
+        )
+
+        params.channelLogger.channel.appendLine(localize(
+            'AWS.samcli.deploy.workflow.packaging',
+            'Packaging SAM Application to S3 Bucket: {0} with profile: {1}',
+            params.deployParameters.packageBucketName,
+            params.deployParameters.profile
+        ))
+        await packageInvocation.execute()
+
+        stage = 'deploying'
+        const deployInvocation = new SamCliDeployInvocation(
```

Comment:
Would it be a good idea to just pass `params.deployParameters` to this  (as well as the other invocations) and just `Pick` the values we need on the other side?

Reply:
I have follow up PRs that should improve the invoke calls.